enterprise/frontend/src/embedding-sdk/cli/utils/get-component-snippets.ts

```diff
@@ -26,33 +35,44 @@ export function getComponentSnippets(options: Options) {
     ? ANALYTICS_PROVIDER_SNIPPET_WITH_USER_SWITCHER
     : ANALYTICS_PROVIDER_SNIPPET_MINIMAL;
 
-  const components = [
+  const components: SampleComponent[] = [
     {
-      name: "analytics-provider",
-      content: analyticsProviderSnippet.trim(),
+      fileName: "analytics-provider",
+      componentName: "AnalyticsProvider",
+      content: withNextJsDirective(analyticsProviderSnippet, isNextJs).trim(),
     },
     {
-      name: "embedding-provider",
+      fileName: "embedding-provider",
+      componentName: "EmbeddingProvider",
       content: embeddingProviderSnippet.trim(),
     },
     {
-      name: "analytics-dashboard",
-      content: analyticsDashboardSnippet.trim(),
+      fileName: "analytics-dashboard",
+      componentName: "AnalyticsDashboard",
+      content: withNextJsDirective(analyticsDashboardSnippet, isNextJs).trim(),
     },
     {
-      name: "theme-switcher",
+      fileName: "theme-switcher",
+      componentName: "ThemeSwitcher",
       content: THEME_SWITCHER_SNIPPET.trim(),
     },
-    {
-      name: "analytics-page",
-      content: ANALYTICS_PAGE_SNIPPET.trim(),
-    },
   ];
 
+  // Only generate the analytics page when not using Next.js.
+  // This is to prevent unintentionally adding multiple providers to pages.
+  if (!isNextJs) {
+    components.push({
+      fileName: "analytics-page",
+      componentName: "AnalyticsPage",
+      content: ANALYTICS_PAGE_SNIPPET.trim(),
+    });
+  }
```

Comment:
The AnalyticsPage component wraps the dashboard with the providers. We don't need in Next.js. In the future I would like to get rid of this too so people import the providers correctly into their App entry point.